build.gradle

```diff
@@ -1,5 +1,8 @@
 apply plugin: 'java'
 apply plugin: "com.github.sherter.google-java-format"
+apply plugin: 'application'
+
+mainClassName = findProperty("main") ?: "com.williamfiset.algorithms.${findProperty("algorithm") ?: 'missingPackage.missingClass'}"
```

Comment:
`gradle run` runs whatever is in the mainClassName variable. We set it to:
1. the value of property 'main', if specified, or
2. "com.williamfiset.algorithms." concatenated with the value of property 'algorithm', if specified, or 'missingPackage.missingClass' otherwise (to cause an error), which will produce an error message (the user did not provide any of the required properties):

```
./gradlew  run 
Error: Could not find or load main class com.williamfiset.algorithms.missingPackage.missingClass

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/home/shared/dev/jdk1.8.0_131/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 786ms
```